File: 120-Triangle/120-Triangle.cpp
// Last updated: 26/11/2025, 22:33:40
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        int row = triangle.size();
        vector<int> memo = triangle[row-1];

        for(int i=row-2; i>=0; i--){
            for(int j=0; j<=i; j++){
                memo[j] = min(memo[j], memo[j+1]) + triangle[i][j];
            }
        }
        return memo[0];
    }
};

File: 120-Triangle/120-Triangle.java
// Last updated: 26/11/2025, 22:33:37
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int size = triangle.size();
        if (size == 1) 
            return triangle.get(0).get(0);

        for (int i = size - 2; i >= 0; i--) {
            for (int j = 0; j < triangle.get(i).size(); j++) {
                int updatedValue = triangle.get(i).get(j) +
                Math.min(triangle.get(i+1).get(j), triangle.get(i+1).get(j+1));
                triangle.get(i).set(j, updatedValue);
            }
        }

        return triangle.get(0).get(0);
    }
}